core/src/main/java/org/keycloak/representations/admin/v2/ClientRepresentation.java

```diff
@@ -0,0 +1,244 @@
+package org.keycloak.representations.admin.v2;
+
+import com.fasterxml.jackson.annotation.JsonInclude;
+import com.fasterxml.jackson.annotation.JsonProperty;
+import com.fasterxml.jackson.annotation.JsonPropertyDescription;
+import jakarta.validation.Valid;
+import jakarta.validation.constraints.NotBlank;
+import jakarta.validation.constraints.NotNull;
+import org.hibernate.validator.constraints.URL;
+import org.keycloak.representations.admin.v2.validation.CreateClient;
+
+import java.util.LinkedHashSet;
+import java.util.Set;
+
+public class ClientRepresentation extends BaseRepresentation {
+
+    public static final String OIDC = "openid-connect";
+
+    @NotBlank(groups = CreateClient.class)
+    @JsonPropertyDescription("ID uniquely identifying this client")
+    private String clientId;
+
+    @JsonPropertyDescription("Human readable name of the client")
+    private String displayName;
+
+    @JsonPropertyDescription("Human readable description of the client")
+    private String description;
+
+    @JsonProperty(defaultValue = OIDC)
+    @JsonPropertyDescription("The protocol used to communicate with the client")
+    private String protocol;
+
+    @JsonPropertyDescription("Whether this client is enabled")
+    private Boolean enabled;
+
+    @URL
+    @JsonPropertyDescription("URL to the application's homepage that is represented by this client")
+    private String appUrl;
+
+    @JsonInclude(JsonInclude.Include.NON_EMPTY)
+    @JsonPropertyDescription("URLs that the browser can redirect to after login")
+    private Set<@NotBlank @URL(message = "Each redirect URL must be valid") String> appRedirectUrls = new LinkedHashSet<String>();
+
+    @JsonInclude(JsonInclude.Include.NON_EMPTY)
+    @JsonPropertyDescription("Login flows that are enabled for this client")
+    private Set<@NotBlank String> loginFlows = new LinkedHashSet<String>();
```

Comment:
Old ClientRepresentation has authenticationFlowBindingOverrides as a map. Think the way it works is that you can overwrite each flow type with a different flow. What this does I don't know, but doesn't seem compatible with the current implementation.

Reply:
On second though I guess this is the enabled flows (authz grants, etc.) for the client. This is probably only related to OIDC though; and we'll probably want to have multiple different client representation formats supported. How will we do that?

If the above assumption is correct then it shouldn't be `loginFlows` as not all flows are login; also it's only considered login if it's an OIDC request, not if it's an OAuth request.